### ISSUE TITLE
🟩 EPIC 6 — Training Loop (epochs, logging, optimizer plug‑in)

### ISSUE THREAD

**Issue by AxelEspinosa94** [AxelEspinosa94/PHOBOS-ML#57]
# Issue 19 — Training Loop (epochs, logging, optimizer plug‑in)

Labels: training enhancement

**Description:**
Implement the full training loop for logistic regression using the existing modules (forward, loss, gradients, optimizer). Add logging for loss per epoch.

**Acceptance Criteria:**

- [ ] train_loop() runs for N epochs
- [ ] Uses optimizer_sgd_step()
- [ ] Logs loss per epoch
- [ ] Tests included
- [ ] Compatible with static/shared library builds
